### ISSUE TITLE
Creating an optimised production build hangs

### ISSUE THREAD

**Issue by minimav** [reflex-dev/reflex#3400]
**Describe the bug**
When I try in run a simple app in production mode within a Docker container the step 'Creating an optimized production build' hangs on step 2/9. When I run the same app in production mode on my machine this step takes <1 minute and the app starts in prod mode as expected. I've observed this hanging for >1 hour.

```
Debug:    Linting and checking validity of types ...
Debug:    ▲ Next.js 14.0.1
Debug: 
Debug:    Creating an optimized production build ...
Creating Production Build:  ━━━━━━━╸                             22% 2/9 0:11:11
```

**To Reproduce**

Single page app based on [counter example in docs](https://reflex.dev/docs/getting-started/introduction/#an-example:-make-it-count) can be used to reproduce along with the following Dockerfile. 

Dockerfile:

```
FROM python:3.10

RUN pip install reflex==0.5.1

COPY rxconfig.py rxconfig.py
COPY app app

EXPOSE 3000
EXPOSE 8000

CMD ["reflex", "run", "--env", "prod", "--loglevel", "debug"]
```

**Expected behavior**
The app should start in a similar timeframe to which is takes when starting outside Docker.

**Specifics (please complete the following information):**
Local setup where this works:
 - Python Version: 3.10.11
 - Reflex Version: 0.5.1
 - OS: macOS Sonoma 14.4.1

Docker setup where it hangs:
- Base image: [Python 3.10](https://hub.docker.com/layers/library/python/3.10/images/sha256-837b16f43b1dcb4cc05e351d3f6f05c07173f8b3a1bf0d4cc02a411e6eca1ea4)
- OS: Debian 11
 
**Additional context**
My suspicion is that I need to do some additional setup in the Dockerfile in terms of installing packages but not clear to me from the debug logging output what that could be. I've had a look at the Dockerfiles in `docker-examples/` e.g. [this one](https://github.com/reflex-dev/reflex/blob/main/docker-example/prod.Dockerfile) but not spotted anything that looks relevant. 

<details>
  <summary>Full log output:</summary>

  ```
  Debug: Config file: '/usr/local/lib/python3.10/site-packages/reflex/config.py'
  Debug: Config: app_name='simple_app' loglevel=<LogLevel.INFO: 'info'> 
  frontend_port=3000 frontend_path='' backend_port=8000 
  api_url='http://localhost:8000' deploy_url='http://localhost:3000' 
  backend_host='0.0.0.0' db_url='sqlite:///reflex.db' redis_url=None 
  telemetry_enabled=True bun_path='/root/.local/share/reflex/bun/bin/bun' 
  cors_allowed_origins=['*'] tailwind={} timeout=120 next_compression=True 
  frontend_packages=[] cp_backend_url='https://rxcp-prod-control-plane.fly.dev' 
  cp_web_url='https://control-plane.reflex.run' 
  gunicorn_worker_class='uvicorn.workers.UvicornH11Worker'
  Warning: The path to the Node binary could not be found. Please ensure that Node
  is properly installed and added to your system's PATH environment variable or 
  try running `reflex init` again.
  Debug: Running command: ['/root/.local/share/reflex/fnm/fnm', '--version']
  Warning: The path to the Node binary could not be found. Please ensure that Node
  is properly installed and added to your system's PATH environment variable or 
  try running `reflex init` again.
  Debug: Running command: ['/root/.local/share/reflex/bun/bin/bun', '-v']
  Debug: [Reflex 0.5.1 with Python 3.10.14 (PATH: /usr/local/bin/python)]
  Debug: [Node None (Expected: 18.17.0) (PATH:None)]
  Debug: [FNM None (Expected: 1.35.1) (PATH: /root/.local/share/reflex/fnm/fnm)]
  Debug: [Bun None (Expected: 1.1.8) (PATH: 
  /root/.local/share/reflex/bun/bin/bun)]
  Debug: [OS Linux Debian GNU/Linux 12 (bookworm)]
  Debug: Using package installer at: /root/.local/share/reflex/bun/bin/bun
  Debug: Using package executer at: None
  Debug: Unzip path: /usr/bin/unzip
  ───────────────────────────────── System Info ──────────────────────────────────
  Debug: Config file: '/usr/local/lib/python3.10/site-packages/reflex/config.py'
  Debug: Config: app_name='simple_app' loglevel=<LogLevel.INFO: 'info'> 
  frontend_port=3000 frontend_path='' backend_port=8000 
  api_url='http://localhost:8000' deploy_url='http://localhost:3000' 
  backend_host='0.0.0.0' db_url='sqlite:///reflex.db' redis_url=None 
  telemetry_enabled=True bun_path='/root/.local/share/reflex/bun/bin/bun' 
  cors_allowed_origins=['*'] tailwind={} timeout=120 next_compression=True 
  frontend_packages=[] cp_backend_url='https://rxcp-prod-control-plane.fly.dev' 
  cp_web_url='https://control-plane.reflex.run' 
  gunicorn_worker_class='uvicorn.workers.UvicornH11Worker'
  Warning: The path to the Node binary could not be found. Please ensure that Node
  is properly installed and added to your system's PATH environment variable or 
  try running `reflex init` again.
  Debug: Running command: ['/root/.local/share/reflex/fnm/fnm', '--version']
  Warning: The path to the Node binary could not be found. Please ensure that Node
  is properly installed and added to your system's PATH environment variable or 
  try running `reflex init` again.
  Debug: Running command: ['/root/.local/share/reflex/bun/bin/bun', '-v']
  Debug: [Reflex 0.5.1 with Python 3.10.14 (PATH: /usr/local/bin/python)]
  Debug: [Node None (Expected: 18.17.0) (PATH:None)]
  Debug: [FNM None (Expected: 1.35.1) (PATH: /root/.local/share/reflex/fnm/fnm)]
  Debug: [Bun None (Expected: 1.1.8) (PATH: 
  /root/.local/share/reflex/bun/bin/bun)]
  Debug: [OS Linux Debian GNU/Linux 12 (bookworm)]
  Debug: Using package installer at: /root/.local/share/reflex/bun/bin/bun
  Debug: Using package executer at: None
  Debug: Unzip path: /usr/bin/unzip
  ─────────────────────────── Initializing simple_app ────────────────────────────
  [13:20:11] Prior to reflex 0.4.0, rx.* components are based on     console.py:95
             Chakra UI. They are now based on Radix UI. To stick to               
             Chakra UI, use rx.chakra.*.                                          
                                                                     console.py:95
             Run `reflex script keep-chakra` to automatically update console.py:95
             your app.                                                            
                                                                     console.py:95
             For more details, please see                            console.py:95
             https://reflex.dev/blog/2024-02-16-reflex-v0.4.0/                    
  Debug: Downloading 
  https://github.com/Schniz/fnm/releases/download/v1.35.1/fnm-linux.zip
  Debug: Downloading https://bun.sh/install
  Debug: Running command: ['bash', '/tmp/tmpd5kzdlka', 'bun-v1.1.8']
  Debug: Installing https://bun.sh/install
  Debug: #=#=#                                                                    
  Debug: ##O#-#                                                                   
  Debug: FNM package downloaded and extracted successfully.
  Debug: Running command: ['/root/.local/share/reflex/fnm/fnm', 'install', 
  '18.17.0', '--fnm-dir', '/root/.local/share/reflex/fnm']
  Debug: Installing node
  Debug: Installing Node v18.17.0 (x64)
  Debug: ##O=#  #                                                                 
  Debug: #=#=-#  #                                                                
  Debug: 
  Debug:                                                                          
  0.0%
  Debug:                                                                          
  1.3%
  ...
  Debug: ######################################################################## 
  100.0%
  Debug: bun was installed successfully to ~/.local/share/reflex/bun/bin/bun 
  Debug: 
  Debug: Added "~/.local/share/reflex/bun/bin" to $PATH in "~/.bashrc" 
  Debug: 
  Debug: To get started, run: 
  Debug: 
  Debug:   source /root/.bashrc 
  Debug:   bun --help 
  [13:20:18] Initializing the web directory.                         console.py:95
  Debug: Setting project hash to 284601405389357886961029775945839507902.
  Debug: Creating .gitignore
  Success: Initialized simple_app
  ───────────────────────────── Starting Reflex App ──────────────────────────────
  Compiling: ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 13/13 0:00:00
  Debug: Running command: ['/root/.local/share/reflex/bun/bin/bun', 'install']
  Debug: Installing base frontend packages
  Debug: bun install v1.1.8 (89d25807)
  Debug:  Resolving dependencies
  Debug:  Resolved, downloaded and extracted [570]
  Debug:  Saved lockfile
  Debug: 
  Debug:  + autoprefixer@10.4.14 (v10.4.19 available)
  Debug:  + postcss@8.4.31 (v8.4.38 available)
  Debug:  + @emotion/react@11.11.1 (v11.11.4 available)
  Debug:  + axios@1.6.0 (v1.7.2 available)
  Debug:  + json5@2.2.3
  Debug:  + next@14.0.1 (v14.2.3 available)
  Debug:  + next-sitemap@4.1.8 (v4.2.3 available)
  Debug:  + next-themes@0.2.1 (v0.3.0 available)
  Debug:  + react@18.2.0 (v18.3.1 available)
  Debug:  + react-dom@18.2.0 (v18.3.1 available)
  Debug:  + react-focus-lock@2.11.3 (v2.12.1 available)
  Debug:  + socket.io-client@4.6.1 (v4.7.5 available)
  Debug:  + universal-cookie@4.0.4 (v7.1.4 available)
  Debug: 
  Debug:  140 packages installed [6.42s]
  Debug: Running command: ['/root/.local/share/reflex/bun/bin/bun', 'add', '-d', 
  'tailwindcss@3.3.2']
  Debug: Installing tailwind
  Debug: bun add v1.1.8 (89d25807)
  Debug:  Resolving dependencies
  Debug:  Resolved, downloaded and extracted [274]
  Debug:  Saved lockfile
  Debug: 
  Debug:  installed tailwindcss@3.3.2 with binaries:
  Debug:   - tailwind
  Debug:   - tailwindcss
  Debug: 
  Debug:  79 packages installed [1.75s]
  Debug: Running command: ['/root/.local/share/reflex/bun/bin/bun', 'add', 
  'lucide-react@0.359.0', '@radix-ui/themes@^3.0.0', 'sonner@1.4.41']
  Debug: Installing frontend packages from config and components
  Debug: bun add v1.1.8 (89d25807)
  Debug:  Resolving dependencies
  Debug:  Resolved, downloaded and extracted [260]
  Debug:  Saved lockfile
  Debug: 
  Debug:  installed lucide-react@0.359.0
  Debug:  installed @radix-ui/themes@3.0.5
  Debug:  installed sonner@1.4.41
  Debug: 
  Debug:  65 packages installed [2.96s]
  Debug: Running command: 
  ['/root/.local/share/reflex/fnm/node-versions/v18.17.0/installation/lib/node_mod
  ules/npm/bin/npm-cli.js', 'run', 'next', 'telemetry', 'disable']
  Debug: Running command: 
  ['/root/.local/share/reflex/fnm/node-versions/v18.17.0/installation/lib/node_mod
  ules/npm/bin/npm-cli.js', 'run', 'export-sitemap']
  Debug: Creating Production Build
  Debug: 
  Debug: > export-sitemap
  Debug: > next build && next-sitemap
  Debug: 
  Debug: Attention: Next.js now collects completely anonymous telemetry regarding 
  usage.
  Debug: This information is used to shape Next.js' roadmap and prioritize 
  features.
  Debug: You can learn more, including how to opt-out if you'd not like to 
  participate in this anonymous program, by visiting the following URL:
  Debug: https://nextjs.org/telemetry
  Debug: 
  Debug:    Linting and checking validity of types ...
  Debug:    ▲ Next.js 14.0.1
  Debug: 
  Debug:    Creating an optimized production build ...
  Creating Production Build:  ━━━━━━━╸                             22% 2/9 0:11:11
  ```
</details>

Thanks and appreciate all your work on the package, it's a pleasure to use.

**Comment by picklelo** [comment]
Not sure if this is the issue, but can you try putting a `reflex init` before you `reflex run` in your Dockerfile?

**Comment by minimav** [comment]
Same behaviour unfortunately. I also tried with the Python 3.11 base image to align with what's in `docker-examples` and had the same issue.

**Comment by picklelo** [comment]
@minimav Perhaps that does make a difference - I just tried your Docker file and built it using `docker build -t <image_name>` and it ran well for me. I wonder if it's because Mac can't run the linux/amd64 image by default. If you try build/run the Dockerfile on a Linux machine does it work?

**Comment by benedikt-bartscher** [comment]
@minimav fyi, I had no problems building multiarch images with buildx

**Comment by MUFC-Andrew-Davies** [comment]
(This is @minimav from my work account)...

Thanks for the feedback and suggestions. Running `docker build -t <image_name>` without the platform argument works for me so it is definitely related to that. We will try this week to build on a Linux machine and see if that's a viable workflow or we have to investigate further to get a build from Mac working.

Since this is not a reflex issue happy to close this or I can wait a short while to update with any resolution we find?

**Comment by picklelo** [comment]
@MUFC-Andrew-Davies I'll close this issue for now - but if you find a fix please post your solution here in case other people run into this in the future.

**Comment by MUFC-Andrew-Davies** [comment]
🙈 This got resolved by updating Docker Desktop. The behaviour described above happened at least up to version 4.22.0, but is fixed in 4.30.0, which is the latest version as of today.
